src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -9248,7 +9336,8 @@ private ErrorPropertySymbol CreateErrorPropertySymbol(ImmutableArray<PropertySym
                     return null;
                 }
             }
-            if (node.SearchExtensionMethods)
+
+            if (node.ReceiverOpt is not BoundTypeExpression && node.SearchExtensionMethods)
```

Comment:
>node.ReceiverOpt is not BoundTypeExpression && [](http://example.com/codeflow?start=16&length=46)

Do we have a unit-test observing effect of this change??

Reply:
Yes. Here are the two tests that prompted this change:

```
✘ Microsoft.CodeAnalysis.CSharp.UnitTests.DelegateTypeTests.ExtensionMethods_03​19ms
Error:
Expected:
                Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "F1").WithLocation(22, 15),
                Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "F2").WithLocation(23, 15)
Actual:
                // (22,15): error CS8917: The delegate type could not be inferred.
                //         d = p.F1;
                Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "F1").WithLocation(22, 15),
                // (23,15): error CS8917: The delegate type could not be inferred.
                //         d = p.F2;
                Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "F2").WithLocation(23, 15),
                // (25,16): error CS8917: The delegate type could not be inferred.
                //         d = E1.F1;
                Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "F1").WithLocation(25, 16),
                // (26,16): error CS8917: The delegate type could not be inferred.
                //         d = E2.F1;
                Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "F1").WithLocation(26, 16)
Diff:
++>                 Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "F1").WithLocation(25, 16)
++>                 Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "F1").WithLocation(26, 16)
Expected: True
Actual:   False

Stack trace:
   at Microsoft.CodeAnalysis.DiagnosticExtensions.Verify(IEnumerable`1 actual, DiagnosticDescription[] expected, Boolean errorCodeOnly) in D:\repos\roslyn\src\Compilers\Test\Core\Diagnostics\DiagnosticExtensions.cs:line 99
   at Microsoft.CodeAnalysis.DiagnosticExtensions.Verify(IEnumerable`1 actual, DiagnosticDescription[] expected) in D:\repos\roslyn\src\Compilers\Test\Core\Diagnostics\DiagnosticExtensions.cs:line 48
   at Microsoft.CodeAnalysis.DiagnosticExtensions.Verify(ImmutableArray`1 actual, DiagnosticDescription[] expected) in D:\repos\roslyn\src\Compilers\Test\Core\Diagnostics\DiagnosticExtensions.cs:line 63
   at Microsoft.CodeAnalysis.DiagnosticExtensions.VerifyDiagnostics[TCompilation](TCompilation c, DiagnosticDescription[] expected) in D:\repos\roslyn\src\Compilers\Test\Core\Diagnostics\DiagnosticExtensions.cs:line 109
   at Microsoft.CodeAnalysis.CSharp.UnitTests.DelegateTypeTests.ExtensionMethods_03() in D:\repos\roslyn\src\Compilers\CSharp\Test\Semantic\Semantics\DelegateTypeTests.cs:line 1867

✘ Microsoft.CodeAnalysis.CSharp.UnitTests.DelegateTypeTests.ExtensionMethods_04​3ms
Error:
Expected:
Actual:
                // (11,15): error CS8917: The delegate type could not be inferred.
                //         d = E.F1;
                Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "F1").WithLocation(11, 15)
Diff:
++>                 Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "F1").WithLocation(11, 15)
Expected: True
Actual:   False

Stack trace:
   at Microsoft.CodeAnalysis.DiagnosticExtensions.Verify(IEnumerable`1 actual, DiagnosticDescription[] expected, Boolean errorCodeOnly) in D:\repos\roslyn\src\Compilers\Test\Core\Diagnostics\DiagnosticExtensions.cs:line 99
   at Microsoft.CodeAnalysis.DiagnosticExtensions.Verify(IEnumerable`1 actual, DiagnosticDescription[] expected) in D:\repos\roslyn\src\Compilers\Test\Core\Diagnostics\DiagnosticExtensions.cs:line 48
   at Microsoft.CodeAnalysis.DiagnosticExtensions.Verify(ImmutableArray`1 actual, DiagnosticDescription[] expected) in D:\repos\roslyn\src\Compilers\Test\Core\Diagnostics\DiagnosticExtensions.cs:line 63
   at Microsoft.CodeAnalysis.DiagnosticExtensions.VerifyDiagnostics[TCompilation](TCompilation c, DiagnosticDescription[] expected) in D:\repos\roslyn\src\Compilers\Test\Core\Diagnostics\DiagnosticExtensions.cs:line 109
   at Microsoft.CodeAnalysis.CSharp.UnitTests.DelegateTypeTests.ExtensionMethods_04() in D:\repos\roslyn\src\Compilers\CSharp\Test\Semantic\Semantics\DelegateTypeTests.cs:line 1895
```